src/app/backend/events.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"time"
+)
+
+// Events response structure.
+type Events struct {
+	// Namespace.
+	Namespace string `json:"namespace"`
+
+	// Event list.
+	Events []Event `json:"events"`
+}
+
+// Single event representation.
+type Event struct {
+	// Message.
+	Message string `json:"message"`
+
+	// Source (following pattern {component host}).
+	Source string `json:"source"`
+
+	// Sub-object.
+	SubObject string `json:"object"`
+
+	// Count.
+	Count int `json:"count"`
+
+	// First seen.
+	FirstSeen string `json:"firstSeen"`
+
+	// Last seen.
+	LastSeen string `json:"lastSeen"`
+
+	// Reason.
+	Reason string `json:"reason"`
+}
+
+// Return events for particular namespace or error if occurred.
+func GetEvents(client *client.Client, namespace string) (*Events, error) {
+	list, err := client.Events(namespace).List(labels.Everything(), fields.Everything())
+
+	if err != nil {
+		return nil, err
+	}
+
+	events := &Events{
+		Namespace: namespace,
+	}
+
+	for _, element := range list.Items {
+		events.Events = append(events.Events, Event{
+			Message:   element.Message,
+			Source:    "{" + element.Source.Component + " " + element.Source.Host + "}",
```

Comment:
Why not both source component and source host. The client may then do whatever they want with this. Preformatting is almost always a bad thing. If you have raw data, send it.